src/pages/gateway/hooks.md

```diff
@@ -0,0 +1,327 @@
+---
+title: Hooks transform
+description: Learn how to use the Hooks transform to add hooks before and after querying your mesh.
+---
+
+# Hooks transform
+
+The Hooks transform allows you to invoke a composable [local or remote](#local-vs-remote-functions) function on a targeted node.
+
+Some use cases for the `HooksTransform` include:
+
+-  Authenticating a user before all operations
+
+-  Publishing events once all operations are executed
+
+-  Creating a cart in a 3rd-party store when calling the `Create Cart` mutation (Adobe Commerce)
+
+<InlineAlert variant="info" slots="text"/>
+
+You cannot use hooks cannot to modify the request or the response. In addition, we recommend that you use resolvers instead of hooks to manipulate data.
```

Comment:
```suggestion
You cannot use hooks to modify the request or the response. In addition, we recommend that you use resolvers instead of hooks to manipulate data.
```